internal/api/verification.go

```diff
@@ -0,0 +1,272 @@
+	Transition()
+	Cancel()
+}
+type verificationStageStart struct {
+	c *VerificationContainer
+}
+
+func (v *verificationStageStart) Transition() {
+	v.c.SendTransition()
+}
+func (v *verificationStageStart) Cancel() {
+	v.c.SendCancel()
+}
+func NewVerificationStageStart(c *VerificationContainer) VerificationStageStart {
+	return &verificationStageStart{c}
+}
+
+type VerificationStageTransitioned interface {
+	Done()
+	VerificationData() VerificationData
+	Approve()
+	Decline()
+	Cancel()
+	Transition()
+}
+type verificationStageTransitioned struct {
+	c *VerificationContainer
+}
+
+func (v *verificationStageTransitioned) Done() {
+	v.c.SendDone()
+}
+func (v *verificationStageTransitioned) VerificationData() VerificationData {
+	return v.c.VData
+}
+func (v *verificationStageTransitioned) Cancel() {
+	v.c.SendCancel()
+}
+func (v *verificationStageTransitioned) Transition() {
+	v.c.SendTransition()
+}
+func (v *verificationStageTransitioned) Approve() {
+	v.c.SendApprove()
+}
+func (v *verificationStageTransitioned) Decline() {
+	v.c.SendDecline()
+}
+func NewVerificationStageTransitioned(c *VerificationContainer) VerificationStageTransitioned {
+	return &verificationStageTransitioned{c}
+}
+
+type VerificationStageDone interface {
+	VerificationState() VerificationState
+}
+type verificationStageDone struct {
+	c *VerificationContainer
+}
+
+func (v *verificationStageDone) VerificationState() VerificationState {
+	return v.c.VState
+}
+func NewVerificationStageDone(c *VerificationContainer) VerificationStageDone {
+	return &verificationStageDone{c}
+}
+
+type VerificationStageCancelled interface {
+}
+
+func NewVerificationStageCancelled(c *VerificationContainer) VerificationStageCancelled {
+	return struct{}{}
+}
+
+type VerificationState string
+
+var (
+	VerificationStateVerified   VerificationState = "verified"
+	VerificationStateUnverified VerificationState = "unverified"
+	VerificationStateUnknown    VerificationState = "unknown"
+)
+
+// VerificationContainer is a helper struct for client implementations.
+// It implements all verification stages, so can be returned at each stage.
+// The container is never exposed directly in tests, as interfaces hide
+// invalid state transitions.
+type VerificationContainer struct {
+	VReq           VerificationRequest
+	VData          VerificationData
+	VState         VerificationState
+	SendReady      func()
+	SendStart      func(method string)
+	SendCancel     func()
+	SendDone       func()
+	SendTransition func()
+	SendApprove    func()
+	SendDecline    func()
+	Mutex          *sync.Mutex
+}
+
+func (c *VerificationContainer) Modify(fn func(cc *VerificationContainer)) {
+	c.Mutex.Lock()
+	defer c.Mutex.Unlock()
+	fn(c)
+}
+
+func (c *VerificationContainer) Stage(stageEnum VerificationStageEnum) VerificationStage {
+	switch stageEnum {
+	case VerificationStageEnumRequested:
+		return NewVerificationStageRequested(c)
+	case VerificationStageEnumRequestedRequetee:
```

Comment:
Is `Requetee` a word I don't know or a typo?

Reply:
yeah it was a typo, will rename to `Recevier`.